client/src/pages/Dashboard.js

```diff
@@ -328,6 +328,7 @@ const Dashboard = () => {
                     }
 
                     localStorage.setItem("dash_previous_chart_data", JSON.stringify(datasets));
+                    localStorage.setItem("dash_previous_criteria", JSON.stringify(criteria))
```

Comment:
Wait so this wasn't there before?

Reply:
I think that by doing the previous feature, I changed the way data_previous_data was read and written by other parts of the code. I tested manually everything to make sure all the concerned features where working as intended. There might still be other issues that we are not aware of at the moment thought.